--- a/ext/extconf.rb	1970-01-01 01:00:00.000000000 +0100
+++ b/ext/extconf.rb	2008-09-28 19:08:38.000000000 +0200
@@ -1,13 +1,13 @@
 
 
 # windows compatibility, need different library name
-if(PLATFORM =~ /mingw|mswin/) then
+if(RUBY_PLATFORM =~ /mingw|mswin/) then
   $libname = '/ms/libpq'
 else
   $libname = 'pq'
 end
 
-if(PLATFORM =~ /darwin/) then
+if(RUBY_PLATFORM =~ /darwin/) then
 	# test if postgresql is probably universal
 	bindir = (IO.popen("pg_config --bindir").readline.chomp rescue nil)
 	filetype = (IO.popen("file #{bindir}/pg_config").
