commit 514da2cec2ef3ca821c444f6b7b24f51bdd6e5d4
Author: Jim Weirich <jim.weirich@gmail.com>
Date:   Wed Jul 10 08:40:04 2013 -0400

    Cleanup for 1.8.7 syntax.

diff --git a/test/test_rake_backtrace.rb b/test/test_rake_backtrace.rb
index 8db3b5a..9a9c9e5 100644
--- a/test/test_rake_backtrace.rb
+++ b/test/test_rake_backtrace.rb
@@ -12,6 +12,8 @@ class TestBacktraceSuppression < Rake::TestCase
 
   def test_system_dir_suppressed
     path = RbConfig::CONFIG['rubylibprefix']
+    skip if path.nil?
+
     paths = [path + ":12"]
 
     actual = Rake::Backtrace.collapse(paths)
@@ -21,6 +23,8 @@ class TestBacktraceSuppression < Rake::TestCase
 
   def test_near_system_dir_isnt_suppressed
     path = RbConfig::CONFIG['rubylibprefix']
+    skip if path.nil?
+
     paths = [" " + path + ":12"]
 
     actual = Rake::Backtrace.collapse(paths)
diff --git a/test/test_rake_clean.rb b/test/test_rake_clean.rb
index a6539fb..1dc7974 100644
--- a/test/test_rake_clean.rb
+++ b/test/test_rake_clean.rb
@@ -15,7 +15,7 @@ class TestRakeClean < Rake::TestCase
     file_name = create_undeletable_file
 
     out, _ = capture_io do
-      Rake::Cleaner.cleanup(file_name, verbose: false)
+      Rake::Cleaner.cleanup(file_name, :verbose => false)
     end
     assert_match(/failed to remove/i, out)
 
@@ -40,7 +40,7 @@ class TestRakeClean < Rake::TestCase
     file_name = File.join(dir_name, "deleteme")
     FileUtils.chmod(0777, dir_name)
     FileUtils.chmod(0777, file_name)
-    Rake::Cleaner.cleanup(file_name, verbose: false)
-    Rake::Cleaner.cleanup(dir_name, verbose: false)
+    Rake::Cleaner.cleanup(file_name, :verbose => false)
+    Rake::Cleaner.cleanup(dir_name, :verbose => false)
   end
 end
diff --git a/test/test_rake_path_map.rb b/test/test_rake_path_map.rb
index b76a949..038ba1f 100644
--- a/test/test_rake_path_map.rb
+++ b/test/test_rake_path_map.rb
@@ -156,8 +156,8 @@ class TestRakePathMap < Rake::TestCase
       "src/org/onstepback/proj/A.java".pathmap("%{src,bin}d/%n.class"))
     assert_equal(
       "src_work/bin/org/onstepback/proj/A.class",
-      "src_work/src/org/onstepback/proj/A.java"
-        .pathmap('%{\bsrc\b,bin}X.class'))
+      "src_work/src/org/onstepback/proj/A.java".
+        pathmap('%{\bsrc\b,bin}X.class'))
     assert_equal(
       ".depends.bak",
       ".depends".pathmap("%X.bak"))
