--- a/Rakefile
+++ b/Rakefile
@@ -4,7 +4,6 @@ $:.unshift 'lib'
 require './lib/hoe.rb'
 
 Hoe.plugin :seattlerb
-Hoe.plugin :isolate
 
 Hoe.spec "hoe" do
   developer "Ryan Davis", "ryand-ruby@zenspider.com"
@@ -14,9 +13,6 @@ Hoe.spec "hoe" do
   blog_categories << "Seattle.rb" << "Ruby"
 
   pluggable!
-  require_rubygems_version '>= 1.4'
-
-  dependency "rake", "~> 0.8" # FIX: to force it to exist pre-isolate
 end
 
 task :plugins do
